craft_application/util/platforms.py

```diff
@@ -0,0 +1,64 @@
+# This file is part of craft_application.
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""OS and architecture helpers for craft applications."""
+from __future__ import annotations
+
+import functools
+import platform
+
+
+@functools.lru_cache(maxsize=1)
```

Comment:
what's the difference between this and `@functools.cache`, since the function itself has no arguments?

Reply:
Hopefully none, but `@functools.cache` is available only in 3.9 onwards.